src/software/world/ball.h

```diff
@@ -159,24 +161,19 @@ class Ball final
      * @param other The Ball to compare against for equality
      * @return True if the other ball is equal to this ball, and false otherwise
      */
-    bool operator==(const Ball& other) const;
+    bool operator==(const Ball &other) const;
 
     /**
      * Defines the inequality operator for a Ball.
      *
      * @param other The ball to compare against for inequality
      * @return True if the other ball is not equal to this ball, and false otherwise
      */
-    bool operator!=(const Ball& other) const;
+    bool operator!=(const Ball &other) const;
 
    private:
-    // All previous positions of the ball, with the most recent position at the front of
-    // the queue, coordinates in meters
-    boost::circular_buffer<Point> positions_;
-    // All previous velocities of the ball, with the most recent velocity at the front of
-    // the queue, coordinates in meters
-    boost::circular_buffer<Vector> velocities_;
-    // All previous timestamps of when the ball was updated , with the most recent
-    // timestamp at the front of the queue, coordinates in meters
-    boost::circular_buffer<Timestamp> last_update_timestamps;
+    // All previous states of the ball, with the most recent position at the front of the
+    // queue, This buffer will never be empty as it's initialized with a BallState on
+    // creation
+    boost::circular_buffer<BallState> states_;
```

Comment:
What if someone constructed the Ball with a `history_size` of 0? Maybe we should specify the buffer_size must be `>= 1`?

Reply:
updated